### ISSUE TITLE
Setting and then resetting numThreads won't work

### ISSUE THREAD

**Issue by asardaes** [RcppCore/RcppParallel#59]
Calling `setThreadOptions()` (i.e. with `numThreads = "auto"`) should call `Sys.unsetenv("RCPP_PARALLEL_NUM_THREADS")`, otherwise, the following happens:

```r
Sys.getenv("RCPP_PARALLEL_NUM_THREADS")
[1] ""
setThreadOptions(2)
Sys.getenv("RCPP_PARALLEL_NUM_THREADS")
[1] "2"
setThreadOptions()
Sys.getenv("RCPP_PARALLEL_NUM_THREADS")
[1] "2"
```

If the internal code always reads this variable, `"auto"` will not have the intended effect after reset.

**Comment by asardaes** [comment]
I don't know if it was intended, but this fix did not make it to v4.3.20.2

**Comment by kevinushey** [comment]
The v4.3.20.2 update was not pushed by us; it was a patch release made by the CRAN maintainers. We're still trying to get an updated version on to CRAN.
